addon/ajax-request.js

```diff
@@ -42,16 +42,17 @@ function isJSONAPIContentType(header) {
   return header.indexOf(JSONAPIContentType) === 0;
 }
 
+let pendingRequestCount = 0;
+
 export default class AjaxRequest {
 
   constructor() {
     this.init();
   }
 
   init() {
-    this.pendingRequestCount = 0;
     if (testing) {
-      Test.registerWaiter(() => this.pendingRequestCount === 0);
+      Test.registerWaiter(() => (pendingRequestCount === 0));
```

Comment:
Yeah, that probably makes more sense. Really good points; might as well do it the "safer" way.